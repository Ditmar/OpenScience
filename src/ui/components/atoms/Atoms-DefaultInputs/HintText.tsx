import React from 'react';
import Typography from '@mui/material/Typography';

export function HintText({ children, error }: { children: string; error?: boolean }) {
  return (
    <Typography
      variant="caption"
      color={error ? 'error' : 'text.secondary'}
      sx={{ marginTop: '4px' }}
    >
      {children}
    </Typography>
  );
}
