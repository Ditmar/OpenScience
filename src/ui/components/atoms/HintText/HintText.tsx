import React from 'react';
import Typography from '@mui/material/Typography';

interface HintTextProps {
  children: string;
  error?: boolean;
}

export function HintText({ children, error }: HintTextProps) {
  return (
    <Typography
      variant="caption"
      color={error ? 'error' : 'text.secondary'}
      sx={{ marginTop: '4px' }}
    >
      {children}
    </Typography>
  );
}
