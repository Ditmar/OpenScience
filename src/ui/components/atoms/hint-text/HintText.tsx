import type { IProps } from './types/IProps';
import styles from './HintText.module.scss';
import { Icon } from '../../../utils/svg-icons/icons';
import infoRaw from '../../../../assets/icons/fi-rr-info.svg?raw';

export default function HintText({ text, size = 'medium', className = '' }: IProps): JSX.Element {
  const sizeClass = styles[`hint-text--${size}`];

  return (
    <div data-testid="hint-text" className={`${styles['hint-text']} ${sizeClass} ${className}`}>
      <Icon
        data-testid="hint-icon-left"
        src={infoRaw}
        className={`${styles['hint-text__icon']} ${sizeClass}`}
      />
      <span className={styles['hint-text__text']}>{text}</span>
      <Icon
        data-testid="hint-icon-right"
        src={infoRaw}
        className={`${styles['hint-text__icon']} ${styles['hint-text__icon--right']} ${sizeClass}`}
      />
    </div>
  );
}
