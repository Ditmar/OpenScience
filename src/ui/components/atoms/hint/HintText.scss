.hint-text {
  display: inline-block;
  margin-top: 4px;

  &--error {
    font-weight: 500;
  }
}
