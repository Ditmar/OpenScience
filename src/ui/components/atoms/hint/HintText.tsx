import React from 'react';
import { Typography } from '@mui/material';
import type { HintTextProps } from './types/IProps';

export function HintText({ text, error = false, id }: HintTextProps) {
  return (
    <Typography id={id} variant="caption" color={error ? 'error' : 'textSecondary'}>
      {text}
    </Typography>
  );
}
