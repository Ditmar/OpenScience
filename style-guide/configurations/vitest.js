const { TEST_FILES } = require('./constants.js');
module.exports = {
  overrides: [
    {
      files: TEST_FILES,
      plugins: ['vitest'],
      extends: ['plugin:vitest/recommended'],
      rules: {
        'import/no-extraneous-dependencies': [
          'off',
          { devDependencies: ['**/*.test.[jt]s?(x)', '**/*.spec.[jt]s?(x)'] },
        ],
      },
    },
  ],
};
