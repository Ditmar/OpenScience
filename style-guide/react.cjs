const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
