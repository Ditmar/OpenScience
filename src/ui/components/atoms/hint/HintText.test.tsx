import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { HintText } from './HintText';

describe('HintText Component', () => {
  it('includes the provided id attribute', () => {
    render(<HintText text="Test hint" id="custom-id" />);
    expect(screen.getByText('Test hint')).toHaveAttribute('id', 'custom-id');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<HintText text="Snapshot test" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
