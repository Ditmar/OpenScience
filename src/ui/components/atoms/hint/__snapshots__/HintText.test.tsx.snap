// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HintText Component > matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="MuiTypography-root MuiTypography-caption css-1jkzbat-MuiTypography-root"
  >
    Snapshot test
  </span>
</DocumentFragment>
`;
